src/GraphQL.Tests/Bugs/Issue4060.cs

```diff
@@ -97,13 +152,14 @@ public override Task BeforeExecutionAsync(IExecutionContext context)
             return Task.CompletedTask;
         }
 
-        private class FieldArgumentDictionary : IReadOnlyDictionary<GraphQLField, IDictionary<string, ArgumentValue>>
+        private class FieldArgumentDictionary : IReadOnlyDictionary<GraphQLField, IDictionary<string, ArgumentValue>>, IDictionary<GraphQLField, IDictionary<string, ArgumentValue>>
```

Comment:
In issue #4060 , the developer wanted to be able to redefine variable values at runtime.  This sample code within `Issue4060.cs` was provided to revert GraphQL.NET v8 to perform variable parsing during execution (vs during validation) as was done in v7, solving his issue.  To properly support merged selection sets, this class must now additionally support `IDictionary<TKey, TValue>` as shown here.  However, the changes shown within this `FieldArgumentDictionary` class are not sufficient to properly support merged selection sets while also allowing runtime changes to variables.